tests/FSharp.Test.Utilities/Compiler.fs

```diff
@@ -0,0 +1,508 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Test.Utilities
+
+open FSharp.Compiler.SourceCodeServices
+open FSharp.Test.Utilities
+open FSharp.Test.Utilities.Assert
+open FSharp.Test.Utilities.Utilities
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.CSharp
+open NUnit.Framework
+open System
+open System.Collections.Immutable
+open System.IO
+
+module Compiler =
```

Comment:
If you make this:  module rec Compiler =      you can lose the and bindings.

module rec
and 
namespace rec 

eliminate the need to have forward only declarations.